src/middlewared/middlewared/plugins/cloud/crud.py

```diff
@@ -71,7 +74,12 @@ async def _validate(self, verrors, name, data):
             if limit1["time"] >= limit2["time"]:
                 verrors.add(f"{name}.bwlimit.{i + 1}.time", f"Invalid time order: {limit1['time']}, {limit2['time']}")
 
-        await self.validate_path_field(data, name, verrors)
+        if self.allow_zvol and (path := await self.get_path_field(data)).startswith("/dev/zvol/"):
+            zvol = zvol_path_to_name(path)
+            if not await self.middleware.call('pool.dataset.query', [['name', '=', zvol], ['type', '=', 'VOLUME']]):
+                verrors.add(f'{name}.{self.path_field}', 'Volume does not exist')
```

Comment:
A zvol underlying an iSCSI target can be in an inconsistent state, unless the operation is being managed by `vcenter_plugin`, so perhaps that should be validated here.

Reply:
What exactly do you want to validate here?

Reply:
Well, my goal is to throw an error if the volume is underlying an iSCSI target and is **not** in an quiesced state.

Since currently we can only be sure if the target is quiesced when  in use by our VM or vmware (via vmware_plugin) we should throw an error for any - let's call them - "non-managed" targets.  (i.e. the ones where the `suspend_vms` or `vmware_sync` below will have no impact).

Reply:
OK, I see you have addressed by the addition of `validate_zpol`.
- `suspend_vms` & `vm.query_snapshot_begin`
- `vmware_sync` & `vmware.dataset_has_vms`